kedro/runner/runner.py

```
@@ -226,38 +226,44 @@ def _run(
         done = None
         max_workers = self._get_required_workers_count(pipeline)
 
-        with self._get_executor(max_workers) as pool:
-            while True:
-                ready = {n for n in todo_nodes if node_dependencies[n] <= done_nodes}
-                todo_nodes -= ready
-                for node in ready:
-                    task = Task(
-                        node=node,
-                        catalog=catalog,
-                        hook_manager=hook_manager,
-                        is_async=self._is_async,
-                        session_id=session_id,
-                    )
-                    if isinstance(pool, ProcessPoolExecutor):
-                        task.parallel = True
-                    futures.add(pool.submit(task))
-                if not futures:
-                    if todo_nodes:
-                        self._raise_runtime_error(todo_nodes, done_nodes, ready, done)
-                    break
-                done, futures = wait(futures, return_when=FIRST_COMPLETED)
-                for future in done:
-                    try:
-                        node = future.result()
-                    except Exception:
-                        self._suggest_resume_scenario(pipeline, done_nodes, catalog)
-                        raise
-                    done_nodes.add(node)
-                    self._logger.info("Completed node: %s", node.name)
-                    self._logger.info(
-                        "Completed %d out of %d tasks", len(done_nodes), len(nodes)
-                    )
-                    self._release_datasets(node, catalog, load_counts, pipeline)
+        pool = self._get_executor(max_workers)
+        if pool is not None:
```

Not sure if it's related to what @merelcht is saying, but for the record I was skimming this code and thought "what if `pool is None`?" (there's no `else` branch here). Just a comment from the peanut gallery.

---

Yeah exactly! I think it might not be immediately obvious that that logic is now inside `SequentialRunner`.

---

Here we could just raise an error saying that abstract `_run` requires executor set.